src/index.tsx

```diff
@@ -659,9 +747,38 @@ export class App extends React.Component<any, AppState> {
     );
   }
 
+  private renderZoomActions() {
+    return (
+      <Stack.Col gap={1}>
+        <Stack.Row gap={1} align="center">
+          {this.actionManager.renderAction(
+            "zoomIn",
+            elements,
+            this.state,
+            this.syncActionResult,
+          )}
+          {this.actionManager.renderAction(
+            "zoomOut",
+            elements,
+            this.state,
+            this.syncActionResult,
+          )}
+          <div style={{ marginLeft: 4 }}>
+            {(this.state.zoom * 100).toFixed(0)}%
+          </div>
+        </Stack.Row>
+      </Stack.Col>
+    );
+  }
+
   public render() {
-    const canvasWidth = window.innerWidth - CANVAS_WINDOW_OFFSET_LEFT;
-    const canvasHeight = window.innerHeight - CANVAS_WINDOW_OFFSET_TOP;
+    const canvasDOMWidth = window.innerWidth - CANVAS_WINDOW_OFFSET_LEFT;
```

Comment:
Just some clean up. It's basically the same than before but with less inline assignments and we can now reuse `canvasScale` for the call to `context.scale`.